From 73a59e93c492389c0d7d17db3b00e21c047cb221 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Mon, 14 Apr 2025 09:15:50 +0200
Subject: [PATCH] Mark `add_tests::add_pypi_git` as an online test
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Mark `add_tests::add_pypi_git` as requiring Internet, as in isolated
environment it fails with:

```
thread 'add_tests::add_pypi_git' panicked at tests/integration_rust/add_tests.rs:915:30:
called `Result::unwrap()` on an `Err` value:   × Request failed after 3 retries
  ├─▶ error sending request for url (https://prefix.dev/conda-forge/noarch/
  │   repodata_shards.msgpack.zst)
  ├─▶ client error (Connect)
  ├─▶ dns error: failed to lookup address information: Name or service not
  │   known
  ╰─▶ failed to lookup address information: Name or service not known
```
---
 tests/integration_rust/add_tests.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/integration_rust/add_tests.rs b/tests/integration_rust/add_tests.rs
index 6cef9054..bb35c795 100644
--- a/tests/integration_rust/add_tests.rs
+++ b/tests/integration_rust/add_tests.rs
@@ -895,6 +895,7 @@ preview = ['pixi-build']"#,
 
 /// Test adding a git dependency using ssh url
 #[tokio::test]
+#[cfg_attr(not(feature = "online_tests"), ignore)]
 async fn add_pypi_git() {
     let pixi = PixiControl::from_manifest(
         format!(
